### PR TITLE
Skip test_specs found inside podspecs

### LINKED ISSUE #894: Jazzy with `.podspec` fails when there is a `test_spec`
I'm using jazzy to document a cocoapod.

In my `.jazzy.yaml`, I specified the pod using `podspec: <My Pod>.podspec`. This used to work fine until I added a `test_spec` to my `.podspec` file:

```
s.test_spec 'Tests' do |test_spec|
  test_spec.source_files = 'Example/Tests/*.{swift}'
end
```

Now there seems to be a second `xcodebuild` phase that tries to build the target `<My-Pod>-Tests`:
```
/Users/<me>/.rvm/gems/ruby-2.4.2/gems/jazzy-0.8.4/lib/jazzy/executable.rb:36:in `execute_command': /Users/<me>/.rvm/gems/ruby-2.4.2/gems/jazzy-0.8.4/bin/sourcekitten ["doc", "--module-name", "<My-Pod>", "--", "-target", "<My-Pod>-Tests", "SWIFT_VERSION\\=\\\"4.0\\\""] (RuntimeError)
```

How can I ignore this `xcodebuild` phase? I don't need my tests to be documented.
Is there any documentation or best practices on how to use jazzy with a `.podspec` file?

### PR DESCRIPTION
**Pull request by johnfairh**
Resolve #894 - ignore CocoaPods test_specs.
Add CLI help for `--podspec`.
Update bundle pins to get hold of appropriate version of CocoaPods.

Tested locally; no specs changes.

### CODE REVIEW

File: lib/jazzy/podspec_documenter.rb [hunk 1]
```diff
@@ -116,6 +116,7 @@ def podfile
                  deterministic_uuids: false
 
         [podspec, *podspec.recursive_subspecs].each do |ss|
+          next if ss.test_specification?
```
Comment (segiddins) [inline lib/jazzy/podspec_documenter.rb]: this will fail for old cocoapods versions that don't have the method

Comment (johnfairh) [reply lib/jazzy/podspec_documenter.rb]: Oh 🤦‍♂️ yes of course.  Thanks.  You think better to `responds_to?` or require cocoapods 1.3 in the gemspec?

Comment (segiddins) [reply lib/jazzy/podspec_documenter.rb]: probably better to add a `responds_to?` for now?

**Comment by johnfairh** [comment]
Tested locally somewhat more thoroughly with combos of CocoaPods version + spec contents.

**Comment by jpsim** [comment]
Thanks for the fix.

**Pull request merged**